shell.nix

```diff
@@ -0,0 +1,12 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+pkgs.mkShell {
+  buildInputs = with pkgs; [
+    ghc
+    cabal-install
+    zlib
+
+    # keep this line if you use bash
+    bashInteractive
+  ];
+}
```

Comment:
Added a `shell.nix` file so it's easy to setup an env, with the required tools/packages. In my case I didn't have GHC or Cabal installed, and the project wouldn't build without `zlib`. This should probably specify the GHC version explicitly, and pin to a `nixpkgs` commit, but I was lazy, and just wanted to get up and running.